doc/reference/networking/lwm2m.rst

```diff
@@ -11,8 +11,281 @@ Overview
+
+       case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_FAILURE:
+           LOG_DBG("Bootstrap registration failure!");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_COMPLETE:
+           LOG_DBG("Bootstrap registration complete");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE:
+           LOG_DBG("Bootstrap transfer complete");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_REGISTRATION_FAILURE:
+           LOG_DBG("Registration failure!");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE:
+           LOG_DBG("Registration complete");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE:
+           LOG_DBG("Registration update failure!");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE:
+           LOG_DBG("Registration update complete");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_DEREGISTER_FAILURE:
+           LOG_DBG("Deregister failure!");
+           break;
+
+       case LWM2M_RD_CLIENT_EVENT_DISCONNECT:
+           LOG_DBG("Disconnected");
+           break;
+
+       }
+   }
+
+Next we sssign `Security` resource values to let the client know where and how
+to connect as well as set the `Manufacturer` and `Reboot` resources in the
+`Device` object with some data and the callback we defined above:
+
+.. code-block:: c
+
+   /*
+    * Server URL of default Security object = 0/0/0
+    * Use leshan.eclipse.org server IP (5.39.83.206) for connection
+    */
+   lwm2m_engine_set_string("0/0/0", "coap://5.39.83.206");
+
+   /*
+    * Security Mode of default Security object = 0/0/2
+    * 3 = NoSec mode (no security beware!)
+    */
+   lwm2m_engine_set_u8("0/0/2", 3);
+
+   #define CLIENT_MANUFACTURER "Zephyr Manufacturer"
+
+   /*
+    * Manufacturer resource of Device object = 3/0/0
+    * We use lwm2m_engine_set_res_data() function to set a pointer to the
+    * CLIENT_MANUFACTURER string.
+    * Note the LWM2M_RES_DATA_FLAG_RO flag which stops the engine from
+    * trying to assign a new value to the buffer.
+    */
+   lwm2m_engine_set_res_data("3/0/0", CLIENT_MANUFACTURER,
+                             sizeof(CLIENT_MANUFACTURER),
+                             LWM2M_RES_DATA_FLAG_RO);
+
+   /* Reboot resource of Device object = 3/0/4 */
+   lwm2m_engine_register_exec_callback("3/0/4", device_reboot_cb);
+
+Lastly, we start the LwM2M RD client (which in turn starts the LwM2M engine).
+The second parameter of `lwm2m_rd_client_start` is the client endpoint name.
+This is important as it needs to be unique per LwM2M server:
+
+.. code-block:: c
+
+   (void)memset(&client, 0x0, sizeof(client));
+   lwm2m_rd_client_start(&client, "unique-endpoint-name", rd_client_event);
+
+Using LwM2M library with DTLS
+*****************************
+
+The Zephyr LwM2M library can be used with DTLS transport for secure
+communication by selecting `CONFIG_LWM2M_DTLS_SUPPORT`.  In the client
```

Comment:
Please add references to the options. So
```
:option:`CONFIG_LWM2M_DTLS_SUPPORT`
```
Please check the rest of the documents if there are other options mentioned.